File: app/src/main/java/com/mert/bilgram/SignUpFragment.java
package com.mert.bilgram;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class SignUpFragment extends Fragment {

    private static EditText emailET, passwordET;
    private static Button signUp;
    private FirebaseAuth mAuth;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_sign_up, container, false);
        emailET = view.findViewById(R.id.mailET);
        passwordET = view.findViewById(R.id.passwordET);
        signUp = view.findViewById(R.id.sign_up_btn);
        mAuth = FirebaseAuth.getInstance();

        signUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                signUp();
            }
        });

        emailET.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                if (charSequence.toString().trim().length() == 0 || passwordET.getText().toString().length() < 6) {
                    signUp.setEnabled(false);
                } else {
                    signUp.setEnabled(true);
                }
            }

            @Override
            public void afterTextChanged(Editable editable) {

            }
        });

        passwordET.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                if (charSequence.toString().trim().length() < 6 || emailET.getText().toString().isEmpty()) {
                    signUp.setEnabled(false);
                } else {
                    signUp.setEnabled(true);
                }
            }

            @Override
            public void afterTextChanged(Editable editable) {

            }
        });

        return view;
    }

    private static void hideKeyboard(Activity activity) {
        InputMethodManager manager = (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
        //Find the currently focused view, so we can grab the correct window token from it.
        View view = activity.getCurrentFocus();
        //If no view currently has focus, create a new one, just so we can grab a window token from it
        if (view == null) {
            view = new View(activity);
        }
        manager.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }

    private void signUp() {

        hideKeyboard(getActivity());

        String email = emailET.getText().toString();
        String password = passwordET.getText().toString();

        mAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(getActivity(), new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {

                        if (task.isSuccessful()) {

                            Intent intent = new Intent(getContext(), FeedActivity.class);
                            startActivity(intent);

                        }

                        Toast.makeText(getContext(), "SIGN UP SUCCESSFUL", Toast.LENGTH_LONG).show();

                    }
                }).addOnFailureListener(getActivity(), new OnFailureListener() {
            @Override
            public void onFailure(@NonNull Exception e) {

                Toast.makeText(getContext(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show();

            }
        });

    }

}

File: app/src/main/java/com/mert/bilgram/SignUpFragment.kt
package com.mert.bilgram

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.auth.FirebaseAuth

class SignUpFragment : Fragment() {
    private var mAuth: FirebaseAuth? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val view = inflater.inflate(R.layout.fragment_sign_up, container, false)
        emailET = view.findViewById(R.id.mailET)
        passwordET = view.findViewById(R.id.passwordET)
        signUp = view.findViewById(R.id.sign_up_btn)
        mAuth = FirebaseAuth.getInstance()
        signUp.setOnClickListener { signUp() }

        emailET.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                signUp.isEnabled = !(charSequence.toString().trim { it <= ' ' }.isEmpty() || passwordET.text.toString().length < 6)
            }

            override fun afterTextChanged(editable: Editable) {}
        })
        passwordET.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                signUp.isEnabled = !(charSequence.toString().trim { it <= ' ' }.length < 6 || emailET.text.toString().isEmpty())
            }

            override fun afterTextChanged(editable: Editable) {}
        })
        return view
    }

    private fun signUp() {
        hideKeyboard(activity!!)
        mAuth!!.createUserWithEmailAndPassword(emailET.text.toString(), passwordET.text.toString())
                .addOnCompleteListener(activity!!) { task ->
                    if (task.isSuccessful)
                        startActivity(Intent(context, FeedActivity::class.java))

                    Snackbar.make(signUp, R.string.sign_up_successful, Snackbar.LENGTH_LONG).show()
                }
                .addOnFailureListener(activity!!) { e -> Toast.makeText(context, e.localizedMessage, Toast.LENGTH_LONG).show() }
    }

    companion object {
        private lateinit var emailET: EditText
        private lateinit var passwordET: EditText
        private lateinit var signUp: Button

        private fun hideKeyboard(activity: Activity) {
            var view = activity.currentFocus
            if (view == null) view = View(activity)

            (activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager).hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}